pkg/resources/rbac.authorization.k8s.io/v1/role/validator.go

```diff
@@ -0,0 +1,85 @@
+package role
+
+import (
+	"fmt"
+
+	"github.com/rancher/webhook/pkg/admission"
+	objectsv1 "github.com/rancher/webhook/pkg/generated/objects/rbac.authorization.k8s.io/v1"
+	"github.com/rancher/webhook/pkg/resources/common"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/trace"
+)
+
+const (
+	grOwnerLabel = "authz.management.cattle.io/gr-owner"
+)
+
+// Validator implements admission.ValidatingAdmissionHandler.
+type Validator struct {
+	admitter admitter
+}
+
+// NewValidator returns a new validator for roles.
+func NewValidator() *Validator {
+	return &Validator{
+		admitter: admitter{},
+	}
+}
+
+// GVR returns the GroupVersionKind for this CRD.
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    "rbac.authorization.k8s.io",
+		Version:  "v1",
+		Resource: "roles",
+	}
+}
+
+// Operations returns list of operations handled by this validator.
+func (v *Validator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{
+		admissionregistrationv1.Update,
+	}
+}
+
+// ValidatingWebhook returns the ValidatingWebhook used for this CRD.
+func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) []admissionregistrationv1.ValidatingWebhook {
+	webhook := admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.NamespacedScope, v.Operations())
+	webhook.ObjectSelector = &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      grOwnerLabel,
+				Operator: metav1.LabelSelectorOpExists,
+			},
+		},
+	}
+	return []admissionregistrationv1.ValidatingWebhook{*webhook}
+}
+
+// Admitters returns the admitter objects used to validate roles.
+func (v *Validator) Admitters() []admission.Admitter {
+	return []admission.Admitter{&v.admitter}
+}
+
+type admitter struct{}
+
+// Admit is the entrypoint for the validator. Admit will return an error if it's unable to process the request.
+// If this function is called without NewValidator(..) calls will panic.
```

Comment:
Is this true? This admitter doesn't seem to use any internal resources like a cache, so we shouldn't have a panic here.